lib/fp.js

```diff
@@ -2,32 +2,29 @@
 
 const { safeCallback } = require('./callbacks');
 
-const partial = (
-  // Partially apply arguments to function
-  fn, // function
-  ...args // array, argumants
-  // Returns: function
-) => (...rest) => fn(...args.concat(rest));
-
-const omap = (
-  // Map object fields with provided function
-  mapFn, // funtion, to appy to every field value
-  obj // object, which fields used for mapping
-  // Returns: object, with same reference but with transformed fields
-) => {
+// Partially apply arguments to function
+//   fn - function
+//   args - array, argumants
+// Returns: function
+//   rest - arguments
+const partial = (fn, ...args) => (...rest) => fn(...args.concat(rest));
+
+// Map object fields with provided function
+//   mapFn - funtion, to appy to every field value
+//   obj - object, which fields used for mapping
+// Returns: object, with same reference but with transformed fields
+const omap = (mapFn, obj) => {
   for (const key in obj) {
     obj[key] = mapFn(obj[key]);
   }
   return obj;
 };
 
-const compose = (
-  // Compose multiple functions into one
-  ...fns // array of functions
-  // Returns: function, composed
-) => (
-  ...args // arguments to be passed to first function
-) => {
+// Compose multiple functions into one
+//   fns - array of functions
```

Comment:
This is not an array, you just pass functions here.

Reply:
`fns` becomes an array after rest operator, so I think we will describe identifier type

Reply:
IMO those descriptions are not code-descriptions, they are usage-descriptions, therefore, they must be written from the point of view of a caller.

Hence here if a caller doesn't know the signature and just read comments/generated docs he may (and should) assume that he must pass an array to this function but (`compose([fn1, fn2,fn3])`) will lead to an incorrect behavior.

I think the best option will be to have it as follows (include rest and description in a doc)
`//    ...fns - functions to be composed`
OR
`//    ...fns - array, functions to be composed`